### PR TITLE
Proofreading and design update

### PR DESCRIPTION
**Pull request by yuxizama**
### Language
Proofreading has been done through all the contents in "Get Started", "Fundamentals", "Guides", and "Tutorials", the main changes are :
- Rephrased sentences to improve readability
- Transform some complex ideas into lists
- Aligned all formats in headings and texts
- Added an introduction at the beginning of each doc
More details can be found in the [WIP Zama Doc Style Guide ](https://www.notion.so/zamaai/Zama-Documentation-Style-Guide-05e27bb48432463b990f3946d9d6ecf8)

⚠️ Note that the proofreading is purely language modifications (as I don't understand the technical details of TFHE-rs). Please review and correct in case my rephrases change some technical implications.

### Design 
I updated the card image design on the landing page based on Rand's requirement [here](https://zama-ai.slack.com/archives/C025KGZJBGC/p1713866735572019).
The main changes are:
- Add icons in the card images
- Unify the card size by adjusting the text in order to harmonize the mobile display

### CODE REVIEW

**Comment by IceTDrinker** [comment]
will rebase to remove the conflicts

File: tfhe/docs/fundamentals/compress.md [hunk 1]
```diff
@@ -75,12 +81,14 @@ fn main() {
...
 
-
 ## Compressed public keys
-This example shows how to compress the classical public keys.
+
+This example shows how to compress the classical public keys:
 
 {% hint style="warning" %}
-It is not currently recommended to use the CompressedPublicKey to encrypt ciphertexts without first decompressing it. In case the resulting PublicKey is too large to fit in memory the encryption with the CompressedPublicKey will be very slow, this is a known problem and will be addressed in future releases.
+It is not currently recommended to use the `CompressedPublicKey` to encrypt ciphertexts without first decompressing them. If the resulting PublicKey is too large to fit in memory, it. may result in significant slowdowns.&#x20;
```
Comment (IceTDrinker) [inline tfhe/docs/fundamentals/compress.md:89]: what is the `&#x20;` ?

File: tfhe/docs/fundamentals/compute.md [hunk 2]
```diff
@@ -1,8 +1,10 @@
 # Compute on encrypted data
 
-Computations on encrypted data should be as easy to write as normal Rust, thanks to the usage of operator overloading.
+This document describes how to perform computation on encrypted data.&#x20;
```
Comment (IceTDrinker) [inline tfhe/docs/fundamentals/compute.md:3]: same question for the &#...

File: tfhe/docs/tutorials/parity_bit.md [hunk 3]
```diff
@@ -188,24 +182,24 @@ fn compute_parity_bit<BoolType>(
...
 where
     BoolType: Clone + Not<Output = BoolType>,
     BoolType: BitXor<BoolType, Output=BoolType>,
 ```
 
-However, the compiler will complain:
+However, the compiler will return an error:
 
-```text
+```
```
Comment (IceTDrinker) [inline tfhe/docs/tutorials/parity_bit.md:202]: causes tests to fail, those annotations are not here by chance, will fix this

File: tfhe/docs/tutorials/parity_bit.md [hunk 4]
```diff
@@ -222,17 +216,17 @@ help: consider extending the `where` bound, but there might be an alternative be
...
 where
     BoolType: Clone + Not<Output = BoolType>,
     &BoolType: BitXor<BoolType, Output=BoolType>,
 ```
 
-The Compiler is still not happy:
+However, this approach still leads to an error:
 
-```text
+```
```
Comment (IceTDrinker) [inline tfhe/docs/tutorials/parity_bit.md:229]: causes tests to fail, those annotations are not here by chance, will fix this

**Comment by IceTDrinker** [comment]
will put other comments, just saw the change making test fails so fixing that first

File: tfhe/docs/fundamentals/decrypt-data.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # Decrypt data
 
-Decrypting data is achieved by using the `decrypt` method, which comes from the FheDecrypt trait.
+This document provides instructions on how to decrypt data.
+
+To decrypt data, use the `decrypt` method from the `FheDecrypt` trait:&#x20;
```
Comment (IceTDrinker) [inline tfhe/docs/fundamentals/decrypt-data.md:5]: &# thing

File: tfhe/docs/fundamentals/compress.md (hunk 1 above)
Comment (yuxizama) [reply tfhe/docs/fundamentals/compress.md:89]: I don't know.... never seen it before. 
Gitbook stuff maybe. I'll correct them

File: tfhe/docs/tutorials/sha256_bool.md [hunk 6]
```diff
@@ -179,9 +203,9 @@ fn xor(a: &[Ciphertext; 32], b: &[Ciphertext; 32], sk: &ServerKey) -> [Ciphertex
 
 #### Addition modulo 2^32
 
-This is perhaps the trickiest operation to efficiently implement in a homomorphic fashion. A naive implementation could use the Ripple Carry Adder algorithm, which is straightforward but cannot be parallelized because each step depends on the previous one.
+This might be the trickiest operation to efficiently implement in a homomorphic manner. An easy implementation could use the Ripple Carry Adder algorithm, which is straightforward but cannot be parallelized because each step depends on the previous one.
```
Comment (IceTDrinker) [inline tfhe/docs/tutorials/sha256_bool.md:206]: no not easy, the right term is naive for algorithms in programming

File: tfhe/docs/fundamentals/serialization.md [hunk 7]
```diff
@@ -68,29 +72,26 @@ fn server_function(serialized_data: &[u8]) -> Result<Vec<u8>, Box<dyn std::error
 }
 ```
 
+## Safe measures
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/fundamentals/serialization.md:75]: Change back to "Safe serialization/deserialization"

File: tfhe/docs/getting_started/security_and_cryptography.md [hunk 8]
```diff
@@ -1,37 +1,38 @@
...
 
 $$plaintext = (\Delta * m) + e$$
 
 $$m \in \mathbb{Z}_p$$
 
-![](../_static/lwe.png)
+![](../\_static/lwe.png)
 
-To go from a **plaintext** to a **ciphertext,** one must encrypt the plaintext using a secret key.
+To transform a **plaintext** into a **ciphertext,** you must encrypt the plaintext using a secret key.
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/getting_started/security_and_cryptography.md:26]: how about `to convert`?

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/docs/getting_started/security_and_cryptography.md:26]: To get a ciphertext from a plaintext ?

File: tfhe/docs/tutorials/parity_bit.md [hunk 9]
```diff
@@ -188,22 +182,22 @@ fn compute_parity_bit<BoolType>(
...
+Next, define the generic bounds with the `where` clause.
 
-In the function, the following operators are used:
+In the function, you can use the following operators:
 
 * `!` (trait: `Not`)
 * `^` (trait: `BitXor`)
 
-By adding them to `where`, this gives:
+Adding them to `where`, the results are as follows:
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/tutorials/parity_bit.md:192]: changed it back

File: tfhe/docs/tutorials/sha256_bool.md [hunk 10]
```diff
@@ -132,19 +146,27 @@ fn sha256(padded_input: Vec<bool>) -> [bool; 256] {
...
+While the structure of the SHA-256 function remains the same, there are some important considerations in the code:
 
-Homomorphic operations are really expensive, so we have to remove their unnecessary use and maximize parallelization in order to speed up the program. To simplify our code we use the Rayon crate which provides parallel iterators and efficiently manages threads.
+* The function signature and the borrowing rules should adapt to the ciphertext type (representing the encrypted bits).
+* Implementing SHA-256 operations with homomorphic encryption uses homomorphic boolean operations internally.
 
-The final code is available at https://github.com/zama-ai/tfhe-rs/tree/main/tfhe/examples/sha256_bool
+Homomorphic operations on encrypted data can be very expansive. Consider these options for better speed:
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/tutorials/sha256_bool.md:158]: you're right

File: tfhe/docs/getting_started/security_and_cryptography.md [hunk 11]
```diff
@@ -45,58 +46,57 @@ $$
...
-Bootstrapping operations are used across the computation to decrease noise within the ciphertexts, preventing it from tampering with the message. The rest of the operations are called leveled because they do not need bootstrapping operations and are usually very fast as a result.
+Noise is critical in _FHE_ because it can tamper with the message if not tracked and managed properly. Bootstrapping operations decrease noise within the ciphertexts and guarantee the correctness of computation. The rest of the operations do not need bootstrapping operations, thus they are called leveled operations and are usually very fast as a result.
 
 The following sections explain the concept of noise and padding in ciphertexts.
 
-### Noise.
+### Noise
 
-For it to be secure, LWE requires random noise to be added to the message at encryption time.
+To ensure security, _LWE_ requires s random noise to be added to the message during encryption.
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/getting_started/security_and_cryptography.md:64]: a lovely typo

File: tfhe/docs/guides/js_on_wasm_api.md [hunk 12]
```diff
@@ -58,12 +58,11 @@ function fhe_uint32_example() {
...
 
-### Web
+## Web
 
-- When using the Web WASM target, there is an additional `init` function to call.
-- When using the Web WASM target with parallelism enabled, there is also one more initialization function to call `initThreadPool`
+For the Web WASM target, call an additional `init` function. With parallelism enabled, call another additional `initThreadPool` function.&#x20;
 
-#### Example
+Example:&#x20;
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/guides/js_on_wasm_api.md:65]: Change to :
"When using the Web WASM target,  you should call an additional init function. With parallelism enabled, you need to call another additional initThreadPool function."

File: tfhe/docs/guides/js_on_wasm_api.md [hunk 13]
```diff
@@ -87,35 +86,31 @@ async function example() {
...
-The sequential browser API and the nodejs API are published as npm packages.
-You can add the browser API to your project using the command `npm i tfhe`.
-You can add the nodejs API to your project using the command `npm i node-tfhe`.
+The browser API and the Node.js API are available as npm packages. Using `npm i tfhe` for the browser API and `npm i node-tfhe` for the Node.js API.
 {% endhint %}
 
 ## Using the JS on WASM API
 
-TFHE-rs uses WASM to expose a JS binding to the client-side primitives, like key generation and encryption, of the Boolean and shortint modules.
+**TFHE-rs** uses WASM to provide a JavaScript(JS) binding to the client-side primitives, like key generation and encryption within the Boolean and shortint modules.
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/guides/js_on_wasm_api.md:103]: hum?

File: tfhe/docs/guides/overflow_operations.md [hunk 14]
```diff
@@ -1,16 +1,20 @@
...
-At the moment, only  additions, subtractions, multiplications are supported. Missing operations will be added soon.
+This document explains how **TFHE-rs** implements specific operations to detect overflows in computations.
 
-The list of operations along with their symbol is:
-| name                                                     | symbol         | type   |
-|----------------------------------------------------------|----------------|--------|
-| [Add](https://doc.rust-lang.org/std/ops/trait.Add.html)  | `overflow_add` | Binary |
-| [Sub](https://doc.rust-lang.org/std/ops/trait.Sub.html)  | `overflow_sub` | Binary |
-| [Mul](https://doc.rust-lang.org/std/ops/trait.Mul.html)  | `overflow_mul` | Binary |
+The mechanism of detecting overflow is to encrypt a flag with a specific ciphertext that reflects the state of the computation. When an overflow occurs, this flag is set to true. Since the server is not able to evaluate this encrypted value, the client has to check the flag value when decrypting to determine if an overflow has happened.
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/guides/overflow_operations.md:5]: How about "The mechanism of detecting overflow is to return an encrypted flag"
It's simpler

File: tfhe/docs/guides/pbs-stats.md [hunk 15]
```diff
@@ -1,12 +1,12 @@
-# PBS Statistics
+# PBS statistics
 
-The `shortint` API now keeps track of how many PBS were executed with a global counter when enabling the `pbs-stats` feature.
+This document explains how to use the PBS statistics feature in **TFHE-rs'**s shortint API to assess the overall computational intensity in FHE applications.
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/guides/pbs-stats.md:3]: no need s here

File: tfhe/docs/guides/js_on_wasm_api.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/docs/guides/js_on_wasm_api.md:103]: sorry, missing a space between javascript and the parenthesis

File: tfhe/docs/guides/zk-pok.md [hunk 16]
```diff
@@ -1,14 +1,14 @@
-# Zero Knowledge proof for Compact Public Key encryption
+# Use zero-knowledge proofs of encryption
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/guides/zk-pok.md:1]: This makes a very long title "Zero-knowledge proofs for the compact public key encryption"
How about just: 

Title: Zero-knowledge proofs
Content: This document explains how to implement the zero-knowledge proofs function **for compact public key encryption** to verify the encryption process without revealing the encrypted information.

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/docs/guides/zk-pok.md:1]: Sounds good

File: tfhe/docs/tutorials/ascii_fhe_string.md [hunk 17]
```diff
@@ -50,7 +46,7 @@ fn to_lower(c: u8) -> u8 {
 }
 ```
 
-We can remove the branch this way:
+To remove the branch, execute:
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/docs/tutorials/ascii_fhe_string.md:49]: yep

File: tfhe/docs/tutorials/parity_bit.md [hunk 18]
```diff
@@ -179,7 +173,7 @@ fn check_parity_bit_validity(
 ) -> bool
 ```
 
-To make it generic, the first step is:
+To make it generic, execute:
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/tutorials/parity_bit.md:176]: changed it back

File: tfhe/docs/tutorials/sha256_bool.md [hunk 19]
```diff
@@ -310,14 +337,14 @@ fn main() {
...
+The program accepts hexadecimal inputs. The input must start with "0x" and contain only valid hex digits, otherwise it will be interpreted as text.
 
-Finally see that padding is executed on the client side. This has the advantage of hiding the exact length of the input to the server, who already doesn't know anything about the contents of it but may extract information from the length.
+Finally， padding is performed on the client side. This has the advantage of hiding the exact length of the input content from the server, thus avoiding the server extracting information from the length, even though the content is fully encrypted.
 
-Another option would be to perform padding on the server side. The padding function would receive the encrypted input and pad it with trivial bit encryptions. We could then integrate the padding function inside the ```sha256_fhe``` function computed by the server.
+It is also feasible to perform padding on the server side. The padding function would take the encrypted input and pad it with a trivial bit of encryption. We can then integrate the padding function into the \`\`sha256\_fhe\`\`\`\` function computed by the server.
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/tutorials/sha256_bool.md:350]: some annoying GitBook style issues

File: tfhe/docs/fundamentals/encrypt-data.md [hunk 20]
```diff
@@ -1,8 +1,10 @@
 # Encrypt data
 
-Encrypting data is achieved via the `encrypt` associated function of the FheEncrypt trait.
+This document explains how to encrypt data.
 
-Types exposed by this crate implement at least one of FheEncrypt or FheTryEncrypt to allow encryption.
+To encrypt data, use the `encrypt` method from the `FheEncrypt` trait. This crate provides types that implement either `FheEncrypt` or `FheTryEncrypt` or both, to enable encryption.
+
+Here is an example:&#x20;
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/fundamentals/encrypt-data.md:7]: what do you mean?

[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/fundamentals/encrypt-data.md:7]: it's an extra blank and doesn't appear &#x20 on Gitbook. I'm removing them

File: tfhe/docs/guides/js_on_wasm_api.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/guides/js_on_wasm_api.md:103]: ah yes, corrected!

File: tfhe/docs/guides/rayon_crate.md [hunk 21]
```diff
@@ -1,16 +1,14 @@
-# Making Rayon And TFHE-RS Work Together
+# Use multi-threading using the rayon crate
```
[the remark this answers is not in the document]
Comment (yuxizama) [reply tfhe/docs/guides/rayon_crate.md]: Seems outdated

**Comment by yuxizama** [comment]
Hey @IceTDrinker 
I've addressed your comments and made some other formatting modifications. All good on my side.
Thanks for the review!

**Review by IceTDrinker — APPROVED** [review]
Looks good thanks !

**Pull request merged**